plugins/cloud-gce/src/main/java/org/elasticsearch/cloud/gce/GceModule.java

```diff
@@ -35,4 +39,46 @@
     protected void configure() {
         bind(GceComputeService.class).to(computeServiceImpl).asEagerSingleton();
     }
+
+    /**
+     * Check if discovery is meant to start
+     *
+     * @return true if we can start gce discovery features
+     */
+    public static boolean isDiscoveryReady(Settings settings, ESLogger logger) {
+        // User set discovery.type: gce
```

Comment:
That would be nice to add a simple unit test for this method and check for different settings combinations.